src/main/java/chess/controller/ChessGame.java

```diff
@@ -0,0 +1,44 @@
+package chess.controller;
+
+import chess.domain.Turn;
+import chess.domain.board.ChessBoard;
+import chess.domain.board.ChessBoardGenerator;
+import chess.dto.BoardStatus;
+import chess.dto.CommandInfo;
+import chess.view.InputView;
+import chess.view.OutputView;
+
+public class ChessGame {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public ChessGame(final InputView inputView, final OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void start() {
+        outputView.printGameStartMessage();
+
+        Turn turn = new Turn();
+        CommandInfo commandInfo = inputView.readCommand();
+        ChessBoard chessBoard = new ChessBoard(ChessBoardGenerator.getInstance());
+
+        if (!commandInfo.command().isStart()) {
+            throw new IllegalArgumentException("아직 게임이 시작되지 않았습니다.");
+        }
+        while (!commandInfo.command().isEnd()) {
```

Comment:
```suggestion
        if (commandInfo.isNotStart()) {
            throw new IllegalArgumentException("아직 게임이 시작되지 않았습니다.");
        }
        while (commandInfo.isNotEnd()) {
```

위처럼 쓰도록 변경될 수 있어보이는데 어떠신가요?!

Reply:
저도 `Not`이라는 단어를 사용해서 메서드 네이밍을 해야할지 많이 고민했었는데요!

`Not`이라는 네이밍을 사용하면 "~일 때" 가 아니라 "~이지 않을 때" 라고 생각을 한 번 더 꺾어서 해야하기 때문에,
안티 패턴이라고 알고 있었고 저도 그 부분에 공감하는 경험이 있었어서 `Not`이라는 네이밍을 사용하지 않았습니다..

알렉스는 이 부분에 대해서 어떤 의견을 가지고 계신지 궁금해요!

---

그리고 궁금한 점이 한 가지 더 있는데요!
`CommandInfo`는 DTO인데, 알렉스의 예시 코드처럼 `isNotEnd()` 등을 구현하려면 DTO 내부에 로직을 작성해야합니다.

```java
public boolean isNotStart() {
    return command != Command.START;
}
```
대충 요런 로직이 될 것 같은데요!
"이 로직은 `Command`에서 처리해야할 도메인 로직으로 느껴지는데, DTO인 `CommandInfo`에 있어도 괜찮을까?" 라는 생각이 듭니다.
저는 DTO는 데이터를 다른 계층으로 전송할 때 사용하는 객체라고만 알고 있는데
이 외에 또 다른 책임을 주어도 괜찮은 것일까요??
이 부분에 대한 알렉스의 생각도 궁금합니다!!

> 추가로 Dto인데 suffix로 Dto를 붙여주지 않아서 어떤 역할을 가졌는 지 이름을 통해 구분이 조금 헷갈릴 수도 있겠다는 생각도 들어서 Dto 클래스들은 뒤에 전부 Dto라는 suffix를 추가했습니다!

Reply:
1. 해당 메소드를 사용할 때 !를 붙여서 사용하는 것과 메소드 네이밍에 Not이 포함되는 것 중 어떤게 더 가독성이 좋아보이는지에 따라 갈릴 것 같아요. 저의 경우 if문 안에 있는 조건문에 하나의 메소드가 아니라 여러 구문이 섞여있을 때 그 if문 안의 조건이 잘 안읽히는 단점이 크게 다가오곤 합니다!

2. DTO에 비즈니스 로직은 최대한 지양하는 것이 좋습니다! 다만, 현재 케이스에서 이게 비즈니스 로직인지도 생각해봐야할 것 같고, 애초에 Command 클래스도 도메인 클래스가 아니지 않나요..? CommnadInfo 클래스의 구조에 대해서 고민해보시면 좋을 것 같아요!

Reply:
> 1. 해당 메소드를 사용할 때 !를 붙여서 사용하는 것과 메소드 네이밍에 Not이 포함되는 것 중 어떤게 더 가독성이 좋아보이는지에 따라 갈릴 것 같아요. 저의 경우 if문 안에 있는 조건문에 하나의 메소드가 아니라 여러 구문이 섞여있을 때 그 if문 안의 조건이 잘 안읽히는 단점이 크게 다가오곤 합니다!

그러면 커멘드가 다음 Step으로 넘어가면 1개가 더 생겨서 isStart(), isNotStart, isEnd, isNotEnd, isMove, isNotMove... 이런식으로 생기게 꽤 많은 메서드가 생기게 될 것 같은데 아래와 같은 메서드는 어떤 것 같으신가요?
```java
    public boolean isType(final Command command) {
        return this == command;
    }

    public boolean isNotType(final Command command) {
        return this != command;
    }
```

---

> 2. DTO에 비즈니스 로직은 최대한 지양하는 것이 좋습니다! 다만, 현재 케이스에서 이게 비즈니스 로직인지도 생각해봐야할 것 같고, 애초에 Command 클래스도 도메인 클래스가 아니지 않나요..? CommnadInfo 클래스의 구조에 대해서 고민해보시면 좋을 것 같아요!

`Command`가 살짝 포지션이 애매하다보니 조금 혼란이 있었네요..! `Command`는 뷰 로직이 맞습니다!
하지만 `Command`가 어떤 열거 상수 종류인지 체크하는 로직은 DTO보다는 `Command` 클래스에서 수행하는 것이 더 적합하다는 생각이 들어서 이렇게 변경해보았는데, 어떻게 생각하시나요!?

```java
// ...
        CommandInfoDto commandInfoDto = inputView.readCommand();
        Command command = commandInfoDto.command();
        ChessBoard chessBoard = new ChessBoard(ChessBoardGenerator.getInstance());

        if (command.isType(Command.START)) {
            throw new IllegalArgumentException("아직 게임이 시작되지 않았습니다.");
        }
        while (command.isNotType(Command.END)) {
// ...
```